src/core/index.ts

```diff
@@ -57,13 +57,26 @@ export const journey = wrapFnWithLocation(
   }
 );
 
-export const step = wrapFnWithLocation(
+type StepType = (name: string, callback: VoidCallback) => Step;
+
+interface StepWithSkipType extends StepType {
```

Comment:
would move it inside step and name it as `StepWithModifiers` or `StepWithAnnotations`.